pkg/apis/networking/types.go

```diff
@@ -235,10 +235,23 @@ type IngressList struct {
 
 // IngressSpec describes the Ingress the user wishes to exist.
 type IngressSpec struct {
-	// A default backend capable of servicing requests that don't match any
-	// rule. At least one of 'backend' or 'rules' must be specified. This field
-	// is optional to allow the loadbalancer controller or defaulting logic to
-	// specify a global default.
+	// IngressClassName is the name of the IngressClass cluster resource. The
+	// associated IngressClass defines which controller will implement the
+	// resource. This replaces the deprecated `kubernetes.io/ingress.class`
+	// annotation. For backwards compatibility, when that annotation is set, it
+	// must be given precedence over this field. The controller may emit a
+	// warning if the field and annotation have different values.
+	// Implementations of this API should ignore Ingresses without a class
+	// specified. An IngressClass resource may be marked as default, which can
```

Comment:
"Implementations of this API should ignore Ingresses without a class specified" is stronger than "Implementations of this API may choose to ignore Ingresses without a class specified" (in the external API doc)

Reply:
I can't find the original feedback around this, but the recommendation was made to move to the stronger language to avoid ambiguity. Thanks for catching the inconsistency, I'll update this everywhere.